web-client/src/components/Modals/ChangeNameModal.tsx

```diff
@@ -0,0 +1,113 @@
+import React, { useEffect } from 'react';
+import { useTranslation } from 'react-i18next';
+
+import { ModalCancelButton, ModalContinueButton } from '../Buttons';
+import { H4Font } from '../figma';
+
+export const ChangeNameModal: React.FC<ChangeNameModalProps> = ({
+  changeNameHandler,
+  showModal,
+  closeModal,
+  initialValues,
+}) => {
+  const { t } = useTranslation();
+  const [form] = Form.useForm();
+
+    useEffect(() => {
+        form.setFieldsValue({
+            displayName: initialValues.displayName,
+            username: initialValues.username,
+        });
+    }, [form, initialValues]);
+
+  return (
+    <Modal
+          visible={showModal}
+          onCancel={closeModal}
+          footer={null}
+      >
+      <>
+        <H4Font>
+          <UserOutlined />
+          {t('settings.changeName')}
+        </H4Font>
+        <Form
+            layout="vertical"
+            form={form}
+            onFinish={values => {
+                changeNameHandler(
+                  values.displayName,
+                  values.username,
+              );
+            }}
+        >
+          <Row gutter={12}>
+            <Col span={24} md={12}>
+              <Form.Item
+                  name="displayName"
+                  rules={[
+                    {
+                      required: true,
+                      message: t(
+                          'settings.changeNameForm.full_name_error_message',
+                      ),
+                    },
+                  ]}
+                  label={t('name')}
+              >
+                <Input
+                    style={{
+                      marginRight: '15px',
+                    }}
+                    placeholder={t('full_name')}
+                />
+              </Form.Item>
+            </Col>
+            <Col span="24" md={12}>
+              <Form.Item
+                  name="username"
+                  label={t('display_name')}
+                  rules={[
+                    {
+                      required: true,
+                      message: t(
+                          'settings.changeNameForm.display_name_error_message',
+                      ),
+                    },
+                  ]}
+              >
+                <Input
+                    placeholder={t('display_name_placeholder')}
+                />
+              </Form.Item>
+            </Col>
+          </Row>
+          <Row>
+            <Col span={11}>
+              <ModalCancelButton onClick={() => closeModal()}>
+                {t('settings.changeNameForm.cancel')}
+              </ModalCancelButton>
+            </Col>
+            <Col span={11} offset={2}>
+              <ModalContinueButton htmlType="submit">
+                {t('settings.changeNameForm.submit')}
+              </ModalContinueButton>
+            </Col>
+          </Row>
+        </Form>
+      </>
+    </Modal>
+  );
+};
+
+interface ChangeNameModalProps {
```

Comment:
Bad interface.  The correct interface for modals is this:
https://github.com/reach4help/reach4help/blob/development/web-client/src/components/Modals/EventTriggeredModal.tsx#L35

This is the interface  for modals
* visibility because visibility is externally triggered
* children so you can put anything inside of it
* buttonPressHandlers:  for a two button "onSubmit" "onCancel" or "onBack" "onForw